src/port1.0/portconfigure.tcl

```diff
@@ -404,12 +404,14 @@ proc portconfigure::configure_start {args} {
 # configure.universal_archs based on supported_archs and build_arch or
 # universal_archs, plus the SDK being used
 proc portconfigure::choose_supported_archs {archs} {
-    global supported_archs configure.sdk_version
+    global supported_archs configure.sdk_version os.major
 
     if {${configure.sdk_version} ne ""} {
         # Figure out which archs are supported by the SDK
         if {[vercmp ${configure.sdk_version} 11.0] >= 0} {
             set sdk_archs [list arm64 x86_64]
+        } elseif {${os.major} == 18 && [file exists /usr/include/sys/cdefs.h]} {
```

Comment:
This check is incomplete; i386 is only supported if `${configure.sdkroot}` is also empty, or equivalently if `${configure.sdk_version}` is the same as `$macosx_version`.

Reply:
I assume I resolved this in the way you explained